types/tx.go

```diff
@@ -44,9 +44,17 @@ type Txs []Tx
 
 // Hash returns the Merkle root hash of the transaction hashes.
 // i.e. the leaves of the tree are the hashes of the txs.
+// TODO(@raneet10): Investigate and decide whether we should handle empty or single tx specially
```

Comment:
Sure. Although as we go ahead for a hardfork, we might not need to calculate tx hashes like we used to in peppermint and just default to how it's done in upstream.  
cc: @sergio-mena @lasarojc

Reply:
I agree that it would be better to have as little divergence from the upstream as possible, if the intent is to drop the fork of cometbft in the future.

Reply:
Agree with @lasarojc, it is important to remove this hunk, otherwise it will be difficult to remove later

Reply:
Resolved (AFAIAC)